DRODLib/Character.cpp

```diff
@@ -1288,6 +1321,43 @@ int CCharacter::parseFactor(const WCHAR *pwStr, UINT& index, CCurrentGame *pGame
 
 		//else: unrecognized identifier -- just return a zero value
 		return val;
+	} else if (pwStr[index] == W_t('#') || pwStr[index] == W_t('@')) {
+		//Find spot where var identifier ends.
```

Comment:
Request: add a comment here describing that this logic is parsing an array index, then looking up the value if it is a valid array variable.